### PR TITLE
Data quality reorganization, taxonomy and sub-facets

### PR DESCRIPTION
**Pull request by stephanegigandet**
This pull request has 3 changes:

1. SiteQuality_off.pm has been split in several modules: DataQuality.pm, DataQualityCommon.pm and DataQualityFood.pm. The new modules have some perlpod documentation. This also removes the need for a symlink from SiteQuality.pm to SiteQuality_off.pm

2. There is a start of a data quality taxonomy. This will be useful in particular to surface quality issues to users (e.g. in the product page for logged in users, or in the product edit form) and to manufacturers (on the producers platform).

3. There are now different levels of issues: bugs, info, warnings and errors. (see bug #2419 )

### CODE REVIEW

File: po/common/common.pot [hunk 1]
```diff
@@ -3688,3 +3688,63 @@ msgstr "Cancel"
...
+msgid "data quality errors"
+msgstr "data quality errors"
+
+msgctxt "data_quality_errors_s"
+msgid "data quality error"
+msgstr "data quality error"
+
+msgctxt "data_quality_errors"
+msgid "data quality errors"
+msgstr "data quality erros"
```
Comment (CharlesNepote) [inline po/common/common.pot]: Small typo: s/erros/errors/

File: po/common/en.po [hunk 2]
```diff
@@ -3564,3 +3564,63 @@ msgstr "Export product data and photos to the public database"
...
+msgid "data quality errors"
+msgstr "data quality errors"
+
+msgctxt "data_quality_errors_s"
+msgid "data quality error"
+msgstr "data quality error"
+
+msgctxt "data_quality_errors"
+msgid "data quality errors"
+msgstr "data quality erros"
```
Comment (CharlesNepote) [inline po/common/en.po]: Small typo: s/erros/errors/

File: po/common/common.pot (hunk 1 above)
Comment (stephanegigandet) [inline po/common/common.pot]:
```suggestion
msgstr "data quality errors"
```

Comment (stephanegigandet) [reply po/common/common.pot]: Fixed, thank you!
BTW did you know that you can make suggestions like that in github? Like in a google doc suggestion, they can be accepted in one click.
image

File: lib/ProductOpener/DataQuality.pm [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+=head1 NAME
+
+ProductOpener::DataQuality - check the quality of data for products
```
Comment (CharlesNepote) [inline lib/ProductOpener/DataQuality.pm]: It does not only check quality but it also stores quality tags inside product data isn't it? Why not separate checking and storing data?

File: po/common/common.pot (hunk 1 above)
Comment (CharlesNepote) [reply po/common/common.pot]: Oh my!... thanks @stephanegigandet !

File: lib/ProductOpener/DataQuality.pm (hunk 3 above)
Comment (stephanegigandet) [reply lib/ProductOpener/DataQuality.pm]: Yes it does store the results of the quality checks as quality tags. But that's the best way to do it I think, having an extra layer between results and storing the results would only add code and complexity. It is also the way all other processing on product data work: we pass a reference to the product object, and it is enriched with new analysis etc.

**Pull request merged**